examples/shapes_and_collections/hatch_demo.py

```diff
@@ -14,23 +23,26 @@
 y1 = np.arange(1, 5)
 y2 = np.ones(y1.shape) * 4
 
-fig, (ax1, ax2, ax3) = plt.subplots(3)
-
-ax1.bar(x, y1, edgecolor='black', hatch="/")
-ax1.bar(x, y2, bottom=y1, edgecolor='black', hatch='//')
-ax1.set_xticks([1.5, 2.5, 3.5, 4.5])
+fig = plt.figure()
+axs = fig.subplot_mosaic([['bar1', 'patches'], ['bar2', 'patches']])
```

Comment:
Happy dance that you're using mosiac (for the record fig, axs = plt.subplot_mosaic also works) 'cause I so think it's well suited for these kinds of things. 
But also not clear why you're reassigning them:
```python
ax = axs['bar1']
ax.bar(x, y2, bottom=y1, edgecolor='black', hatch=['*', 'o', 'O', '.'])
```
versus using 'em directly:
```python
axs['bar'].bar(x, y2, bottom=y1, edgecolor='black', hatch=['*', 'o', 'O', '.'])
```

Reply:
Less typing?

Reply:
I don't think it's that much less and I think the reassignment especially w/ same variable name is  potential unnecessary source of failure.

Reply:
... wrote out the long form

Reply:
>  I think the reassignment especially w/ same variable name is potential unnecessary source of failure.

I'd argue the opposite - its less prone to typos and easier to refactor.

Reply:
I agree it's easier to refactor, but it's prone to leaving out that assignment statement and therefore messing up all the references.